Files/warranty_ibm_lenovo.py

```diff
@@ -93,16 +98,20 @@ def process_result(self, result, purchases):
         global full_serials
         data = {}
 
-        for item in result:
+        # The API returns results for single devices in a different format than multiple devices, this keeps everything
+        # returned from the API in the same list format
+        if 'Warranty' in result:
+            result = [result]
 
+        for item in result:
             # Warranties
             if 'Warranty' in item and len(item['Warranty']) > 0:
                 warranties = item['Warranty']
             else:
                 continue
 
             data.clear()
-            serial = item['ID']
+            serial = item['Serial']
```

Comment:
It doesn't look like this was even being used.

Reply:
Yea, since we were waiting to get the script validated the changes to the script for the new API were based solely on the API docs. The docs stated that the following items would be returned. Serial number was not mentioned so I assumed this was the serial number was listed as ID. 

InWarranty - Whether this product is in warranty
Purchased - Date this product was purchased Will not be available if we do not have the purchase information (product was not registered)
Shipped - Date this product was shipped
Warranty - List of warranties
ID
Type
Name
Start
End
Contract - List of contract warranties. Available only if the product is part of a contract
Contract
Quantity
ItemNumber
ChargeCode
SLA
EntitlementCode
Status
Start
End